cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.md

```diff
@@ -523,22 +523,141 @@ public class CSRFValidationFilter implements Filter {
      * @param httpRequest  Source HTTP request
      * @param httpResponse HTTP response object to update
      */
-    private void addTokenCookieAndHeader(HttpServletRequest httpRequest, 
+    private void addTokenCookieAndHeader(HttpServletRequest httpRequest,
                                          HttpServletResponse httpResponse) {
         //Get new token
         String token = this.generateToken();
-        //Add cookie manually because the current Cookie class implementation 
+        //Add cookie manually because the current Cookie class implementation
         //do not support the "SameSite" attribute
         //We let the adding of the "Secure" cookie attribute to the reverse proxy rewriting...
         //Here we lock the cookie from JS access and we use the SameSite new attribute protection
-        String cookieSpec = String.format("%s=%s; Path=%s; HttpOnly; SameSite=Strict", 
+        String cookieSpec = String.format("%s=%s; Path=%s; HttpOnly; SameSite=Strict",
                             this.determineCookieName(httpRequest), token, httpRequest.getRequestURI());
         httpResponse.addHeader("Set-Cookie", cookieSpec);
         //Add cookie header to give access to the token to the JS code
         httpResponse.setHeader(CSRF_TOKEN_NAME, token);
     }
 }
 ```
+# JavaScript Guidance for Auto-inclusion of CSRF tokens as an AJAX Request header
+
+The following guidance considers **GET**, **HEAD** and **OPTIONS** methods are safe operations. Therefore **GET**, **HEAD**, and **OPTIONS** method AJAX calls need not be appended with a CSRF token header. However, if the verbs are used to perform state changing operations, they will also require a CSRF token header.
+
+**POST**, **PUT**, **PATCH**, **DELETE** and **TRACE** verbs, being state changing verbs, should have a CSRF token attached to the request. The following guidance will demonstrate how to create overrides in JavaScript libraries to have CSRF tokens included automatically with every AJAX request for the state changing methods mentioned above.
+
+## Storing the CSRF Token Value in the DOM
+
+A CSRF token can be included in the <meta> tag as shown below. All subsequent calls in the page can extract the CSRF token from this <meta> tag. It can also be stored in a JS variable / anywhere on the DOM. However, it is not recommended to store it in cookies or browser local storage.
```

Comment:
Explain why it is not recommended to store it in cookies or browser local storage.
I understand that this is not recommended because of the reasons that were explained before [in double submit cookie](https://github.com/OWASP/CheatSheetSeries/blob/master/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.md#double-submit-cookie) - if so please refer/link to it.